### ISSUE TITLE
Temporary local storage entries are not cleared when token request has "state" value

### ISSUE THREAD

**Issue by AndreiZe** [AzureAD/microsoft-authentication-library-for-js#1333]
## Library
- [msal@1.2.1] 

## Framework
If you are using a framework, please provide the framework and version (e.g. Angular v8, React v16, etc).
React 16.8.1

## Description
Local storage use grows uncontrollably when silent token fetch request has "state" value passed as part of AuthenticationParameters object. Eventually it results in localStorage exhaustion and failure to fetch tokens in case localStorage is used to cache tokens locally. There is no simple recovery except for calling logout().

## Security
No

## Regression
Not sure

## Configuration
Please provide your MSAL configuration options.

```js
// Provide configuration values here.
// For Azure B2C issues, please include your policies.
      cache: {
        cacheLocation: 'localStorage',
        storeAuthStateInCookie: false
      },
```

## Expected behavior
MSAL should automatically clean up any temporary local storage entries which it uses to facilitate silent token fetch. it fails to do so when "state" value passed as part of AuthenticationParameters object. 

When token request has state then MSAL use it as key which is embedded into local storage keys which are used to store temporary state. The key is generated in the following format {guid}|{state} where guid is generated by MSAL.

The problem is in implementation of resetTempCacheItems method:

![image](https://user-images.githubusercontent.com/7536650/75739573-14bfad80-5cba-11ea-85ed-2e0839ab5893.png)

More specifically, line 138 determine key state based on last occurrence of '|' symbol but does not account for the fact that state itself has this symbol inside it.

## Browsers
Is this issue browser-specific? If so, please detail which browsers are impacted (e.g. IE 11, Safari).
Not browser specific

**Comment by sameerag** [comment]
Thanks @AndreiZe, this is an important find and will fix it asap. Please check here for the PR link.

**Comment by sameerag** [comment]
PR #1339 should fix this.

**Comment by jasonnutter** [comment]
The fix for this is available in `msal@1.2.2-beta.2`, please try that version and confirm that it has been fixed, thanks!
